### ISSUE TITLE
[Data Grid] Make `SUBMIT_FILTER_STROKE_TIME` editable via props

### ISSUE THREAD

**Issue by managervcf** [mui/mui-x#9157]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

I would love to be able to control the debounce time on applied filters inside the filterPanel. Currently [it's hardcoded here](https://github.com/mui/mui-x/blob/bd291b3ab9032c8259bdbf78ff1be40b616fad74/packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterInputValue.tsx#L8).

Alternatively, please allow users to be able to apply filters manually - via some kind of `Apply filters` button.

### Motivation 🔦

This would allow me to optimize the Data Grid with server-side filtering and reduce the amount of unnecessary back-end requests.

**Comment by MBilalShafi** [comment]
You can [customize the filter operators](https://mui.com/x/react-data-grid/filtering/#multiple-values-operator) to change the `InputComponent` to your custom one, inside the component, you can set the filter stroke time as you want.
The easiest way is to reuse the code of [GridFilterInputValue](https://github.com/mui/mui-x/blob/bd291b3ab9032c8259bdbf78ff1be40b616fad74/packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterInputValue.tsx) component and use it as a custom `InputComponent`.

Here's a demo where stroke time is being used: https://mui.com/x/react-data-grid/filtering/#multiple-values-operator.

> Alternatively, please allow users to be able to apply filters manually - via some kind of Apply filters button.

Can you explain more why you need this, the filters are applied automatically after the stroke time has passed, you can customize the `InputComponent` of the filter operator to do this immediately by calling [`applyValue`](https://github.com/mui/mui-x/blob/bd291b3ab9032c8259bdbf78ff1be40b616fad74/packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterInputValue.tsx#L25) function passed in props.

**Comment by managervcf** [comment]
>You can [customize the filter operators](https://mui.com/x/react-data-grid/filtering/#multiple-values-operator) to change the InputComponent to your custom one, inside the component, you can set the filter stroke time as you want.
The easiest way is to reuse the code of [GridFilterInputValue](https://github.com/mui/mui-x/blob/bd291b3ab9032c8259bdbf78ff1be40b616fad74/packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterInputValue.tsx) component and use it as a custom InputComponent.

>Here's a demo where stroke time is being used: https://mui.com/x/react-data-grid/filtering/#multiple-values-operator.

Of course creating a custom `InputComponent` is something that I would have to do in the end if the `SUBMIT_FILTER_STROKE_TIME` remains unaccessible.

However, I feel that enabling this as a prop would be a better solution - like the `debounceMs` prop inside the [GridToolbarQuickFilter](https://github.com/mui/mui-x/blob/35ff68b3aad3c0963bfa656b8c56a464011bbfa4/packages/grid/x-data-grid/src/components/toolbar/GridToolbarQuickFilter.tsx#L67).

**Comment by romgrk** [comment]
I think we should expose it.

@MBilalShafi do you have any suggestion under which name we could expose it? Should it go on the root props or elsewhere?

**Comment by MBilalShafi** [comment]
@romgrk Yeah, it'll be less painful if it's readily customizable without customizing filter operators.

Probably, `slotProps.filterPanel.filterDebounceMs` could be a better option, then, we will have to do something like this in the respective components (e.g. `GridFilterInputValue`):
```tsx
const debounceMs = rootProps.slotProps.filterPanel.filterDebounceMs ?? DEFAULT_FILTER_STROKE_TIME
```
And we'll also have to exclude this prop somehow when we pass down the `...slotProps.filterPanel` to the `GridFilterPanel` component.
https://github.com/mui/mui-x/blob/33411d878209767c53e0d2156cbc1f31937c54dc/packages/grid/x-data-grid/src/hooks/features/filter/useGridFilter.tsx#L379

What do you think?

**Comment by romgrk** [comment]
Is that setting only related to the filter panel? Could we make it a prop of the filter panel so we don't have to worry about excluding?

**Comment by MBilalShafi** [comment]
Yeah, we can make it a part of `filterFormProps` but the problem with that is if we pass it generically in `currentOperator.InputComponent` as a prop, it may cause _React does not recognize the x prop on a DOM element_ error for the `InputComponents` that do not use it. e.g `GridFilterInputBoolean` or GridFilterInputSingleSelect` as they destructure `{ ...others }` to the underlying components.

https://github.com/mui/mui-x/blob/13658f04f74d4e693b3ed5d9cb0408c15675ece6/packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterForm.tsx#L538-L545

So, the possible options seem to be, either we stop destructuring the `others` object in the internal components and make it a part of `filterFormProps` or use something like `slotProps.filterPanel.filterDebounceMs` as I suggested in the comment above.
